### ISSUE TITLE
Errors in `gdbus_codegen` annotations support

### ISSUE THREAD

**Issue by inigomartinez** [mesonbuild/meson#9883]
**Describe the bug**
There are changes in `gdbus_codegen` that make the build to fail because it now doesn't accept empty lists as possible input.

**To Reproduce**
This happens with [ModemManager](https://gitlab.freedesktop.org/mobile-broadband/ModemManager/-/commit/86496bd326f2ec139101049fb19a10b255ee8ad5). An example would be as follows:
```
  gnome.gdbus_codegen(
    'example',
    sources: 'example.c',
    interface_prefix: 'org.example.',
    namespace: 'Example',
    annotations: [],
    object_manager: false,
    autocleanup: 'objects',
    build_by_default: true,
  )
```

**Expected behavior**
The annotation parameter should be ignored if empty list is provided.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
* what Python version are you using e.g. 3.8.0
* what `meson --version`
* what `ninja --version` if it's a Ninja build
This issue starts at meson 0.61.

**Comment by eli-schwartz** [comment]
Why should this be categorized as a regression instead of "better input sanity-checking"?

**Comment by inigomartinez** [comment]
I have categorized it as a regression because it made [already working builds to fail](https://gitlab.freedesktop.org/mobile-broadband/ModemManager/-/issues/499#note_1232380).

However, I can change the description if you think it fits better.

**Comment by eli-schwartz** [comment]
"made already working builds to fail" is not the definition of a regression, though.

A regression is when already working builds ***that could reasonably be expected to work*** start failing.

More specifically, a regression is a subclass of a bug. If something is not a bug, then it doesn't matter whether previously working builds now fail -- it becomes the responsibility of the people with failing builds, to fix *their* definitions.

For example, when i18n.merge_file started erroring out with "function does not take positional arguments", that wasn't a regression, even though it caused already working builds to fail, because the builds were always wrong. That's why meson 0.61.0 will not back out the change, even though we did for 0.60 and very few people actually took advantage of the deprecation warning, because those builds deserve to be broken and we aren't carrying a long-term workaround.

So again I ask my original question: why did you need to pass an empty list, when semantically it is supposed to accept a list of 3 elements. e.g. can't you just... Not specify the kwarg at all?

I'm not saying this isn't a bug. I'm not saying it *is* a bug either. I'm just asking you to elaborate on the problem... no need to play word games with the description...

**Comment by eli-schwartz** [comment]
> similar approach where they have a list of targets where some of them create annotations and others don't

Thank you for clarifying the context in which this came up. :)

@dcbaker, what do you think?

**Comment by dcbaker** [comment]
I think we should allow an empty list. This would have worked in 0.61.0, it was our fixes in 0.61.1 that broke it.
